### ISSUE TITLE
when using detector.py the result is all bigger than 0.91

### ISSUE THREAD

**Issue by shipleyxie** [AravindGanesh/VAD#1]
after using "pyton vad_model " i got a .h5 learned model ,train accuracy is 0.8491, while using detector.py to a audio which contain not speech ,and the restult is all bigger than 0.9, mean=0.9 std=0.04.
could You please give some guide to solve this problem? @AravindGanesh

**Comment by AravindGanesh** [comment]
This code was written in my beginning days of DL and there are a few basic mistakes. 
Firstly the `vad_model.py` is a simple MLP - it does not work well with time-series data. Here the problem is to detect whether a window of a sound file is human voice or noise. 
I suggest you to use `rnn_model.py` instead. 
+ Download some opensource speech dataset online. They are usually clean (without noise). 
+ Use `feature_extractor.py` to extract 13 dimensional mfcc features of the windows. 
 In `feature_extractor.py` we note down the time intervals of speech and silence, add noise at some SNR (value of your choice, do experiments with different values). Get mfcc features vectors of the samples - and normalize them (or just batch norm layers in your model),  and label feature vectors for speech+noise as 1 and label feature vectors for noise (silence+noise) as 0 and saves the feature vectors in CSV files. 
+ in `rnn_model.py` change the activations of layers to `relu'  (two layers with linear activation are equivalent to 1 a single layer since each layer is just matrix multiplication - this was a naive mistake I did in my beginning days xD)
+ refer to 'detector.py` - to load model and test your model. 

I have worked on this project over a year ago. I will try to update the code to the latest version of tensorflow, tf.keras and tf.data.Dataset API instead of saving to CSV.

@shipleyxie , you may also try using a network with 1D convolution layers.

**Comment by AravindGanesh** [comment]
Also, another correction in `feature_extractor.py` 
To extract MFCC features, use the following lines of code. 
```
def get_mfcc_features(y):
  feats = librosa.feature.mfcc(
                      y=y, 
                      sr=sr, 
                     n_mfcc=13,
                      n_fft=int(0.025*sr), # window length 25ms 
                      hop_length=int(0.015*sr) # stride=15ms,  or overlap = 10ms
                       )
  return feats.T  # array of shape (x, 13): each 13 dimensional vector is your input.
``` 
take transpose and stack the 13 dimensional vectors.
